Domain/Logic/Deserialize/Features.cs

```diff
@@ -0,0 +1,216 @@
+                    runtimeFeatures = JsonConvert.DeserializeObject<Mvc.Microservices.Feature[]>(await url.Download());
+                }
+                catch (Exception ex)
+                {
+                    Log.For(typeof(Features)).Warning(ex.ToString());
+                }
+                await Repository.Write($"/features/services/{service.Name}.json", JsonConvert.SerializeObject(runtimeFeatures));
+            }
+            await RefreshFeatures();
+        }
+        internal async static Task RefreshFeatures()
+        {
+            var features = GetXmlFeatures().ToList();
+            foreach (var service in Service.All)
+            {
+                try
+                {
+                    var batch = JsonConvert.DeserializeObject<FeatureDefinition[]>(await Repository.Read($"/features/services/{service.Name}.json"));
+                    batch.Do(x => x.For(service));
+                    features.AddRange(batch);
+                }
+                catch (Exception ex)
+                {
+                    Log.For(typeof(Features)).Warning(ex.ToString());
+
+                }
+            }
+            await Repository.Write("/features/features.json", JsonConvert.SerializeObject(features));
+            Feature.All = await LoadFeatures();
+        }
+        internal static async Task<Feature[]> LoadFeatures()
+        {
+            var json = "";
+            try
+            {
+                json = await Repository.Read("/features/features.json");
+            }
+            catch (Exception ex)
+            {
+                Log.For(typeof(Feature)).Error(ex.ToString());
+                await RefreshServiceFeatures();
+            }
+            if (json.HasValue())
+            {
+                var featureDefinitions = JsonConvert.DeserializeObject<FeatureDefinition[]>(json);
+                return GetActualFeatures(featureDefinitions);
+            }
+            return Feature.All.ToArray();
+        }
+        static Feature[] GetActualFeatures(FeatureDefinition[] featureDefinitions)
+        {
+            var features = new List<Feature>();
+            foreach (var defenition in featureDefinitions)
+            {
+                Feature parent = null;
+                foreach (var title in defenition.FullPath.Split("/"))
+                {
+                    var feature = features.FirstOrDefault(x => x.Title.CompareTo(title) == 0);
+                    if (feature != null)
+                    {
+                        parent = feature;
+                        continue;
+                    }
+                    var fullPath = title;
+                    if (parent != null) fullPath = parent.GetFullPathSlashSeperated() + "/" + title;
+                    var subfeatureDefenition = featureDefinitions.FirstOrDefault(x => x.FullPath.CompareTo(fullPath) == 0);
+                    if (defenition.FullPath.ToLower().EndsWith(title.ToLower())) feature = CreateFeature(defenition, parent);
```

Comment:
Use this before the loop, and use that here (.Last())
var pathParts = definition.FullPath.Split('/')